chain-evm/src/machine.rs

```diff
@@ -0,0 +1,248 @@
+//! # The Virtual Machine
+//!
+//! Abstractions for the EVM runtime.
+//!
+//! This module contains types that abstract away implementation details from the `evm` crate,
+//! specially those involving node and stack configurations, and runtime execution.
+//!
+//! ## Handler <- EVM Context Handler
+//! ## StackState<'config>
+//!
+
+use std::rc::Rc;
+
+use evm::{
+    backend::{Apply, ApplyBackend, Backend, Basic, Log, MemoryVicinity},
+    executor::{MemoryStackState, StackExecutor, StackSubstateMetadata},
+    Config, Context, Runtime,
+};
+use primitive_types::{H160, H256, U256};
+
+use crate::state::AccountTrie;
+
+/// Environment values for the machine backend.
+pub type Environment = MemoryVicinity;
+
+/// The context of the EVM runtime
+pub type RuntimeContext = Context;
+
+/// Top-level abstraction for the EVM with the
+/// necessary types used to get the runtime going.
+pub struct VirtualMachine {
+    environment: Environment,
+    state: AccountTrie,
+    logs: Vec<Log>,
+}
+
+impl VirtualMachine {
+    /// Creates a new `VirtualMachine` given environment values and account storage.
+    pub fn new(environment: Environment, state: AccountTrie) -> Self {
+        Self {
+            environment,
+            state,
+            logs: Default::default(),
+        }
+    }
+    /// Returns an initialized instance of `evm::executor::StackExecutor`.
+    pub fn executor<'backend, 'config>(
+        &'backend self,
+        gas_limit: u64,
+        config: &'config Config,
+    ) -> StackExecutor<'config, MemoryStackState<'backend, 'config, VirtualMachine>> {
```

Comment:
So we expose `StackExecutor` from `evm` in a public method. What is this method used for?

Reply:
I just added this method to be able to get an executor running, there's no need for it to be public. I've been waiting for feedback on how to proceed.